go/vt/vitessdriver/driver_go18.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2016, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build go1.8
+
+// TODO(sougou): Merge this with driver.go once go1.7 is deprecated.
+// Also write tests for these new functions once go1.8 becomes mainstream.
+
+package vitessdriver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+)
+
+const noIntermixingErr = errors.New("named and positional arguments intermixing disallowed")
+
+func (c *conn) ExecContext(ctx context.Context, query string, args []driver.Value) (driver.Result, error) {
+	if c.Streaming {
+		return nil, errors.New("Exec not allowed for streaming connections")
+	}
+
+	bv, err := bindVarsFromNamedValues(args)
+	if err != nil {
+		return nil, err
+	}
+	qr, err := c.exec(ctx, query, bv)
+	if err != nil {
+		return nil, err
+	}
+	return result{int64(qr.InsertID), int64(qr.RowsAffected)}, nil
+}
+
+func (c *conn) QueryContext(ctx context.Context, query string, args []driver.Value) (driver.Rows, error) {
+	bv, err := bindVarsFromNamedValues(args)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.Streaming {
+		stream, err := c.vtgateConn.StreamExecute(ctx, query, bindVars, c.tabletTypeProto, nil)
+		if err != nil {
+			return nil, err
+		}
+		return newStreamingRows(stream, nil), nil
+	}
+
+	qr, err := c.exec(ctx, query, bv)
+	if err != nil {
+		return nil, err
+	}
+	return newRows(qr), nil
+}
+
+func (s *stmt) ExecContext(ctx context.Context, args []driver.Value) (driver.Result, error) {
+	return s.c.ExecContext(ctx, s.query, args)
+}
+
+func (s *stmt) QueryContext(ctx context.Context, args []driver.Value) (driver.Rows, error) {
+	return s.c.QueryContext(ctx, s.query, args)
+}
+
+func bindVarsFromNamedValues(args []driver.NamedValue) (map[string]interface{}, error) {
+	bv := make(map[string]interface{}, len(args))
+	nameUsed := false
+	for i, v := range args {
+		if i == 0 {
+			// Determine if args are based on names or ordinals.
+			if args.Name != "" {
+				nameUsed = true
+			}
+		} else {
+			// Verify that there's no intermixing.
+			if nameUsed && args.Name == "" {
+				return nil, noIntermixingErr
+			}
+			if !nameUsed && args.Name != "" {
+				return nil, noIntermixingErr
+			}
+		}
+		if args.Name == "" {
+			bv[fmt.Sprintf("v%d", i+1)] = v
+		} else {
+			if args.Name[0] == ":" || args.Name[0] == "@" {
```

Comment:
Probably append to user docs that no symbol prefix and if provided it will be stripped.